### DISCUSSION TITLE
How can I use a trigger function to update the `profiles` table every time the `auth.users` table is updated?

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#8087] [Questions]
Hi! I'm creating a new profile in my public `Profiles` table every time the user signs up with Supabase Auth. I'm doing that using the following trigger function:

```
begin
  insert into public."Profile"(id, email)
  values(new.id, new.email);

  return new;
end;
```

Now I need to write a function that updates the profile's email every time the `auth.user`'s email is updated, and a function that deletes the profile every time the `auth.user` is deleted. Can you please help me to figure out how to do that?

Also, where can I find the tutorials/docs that can teach me how to do stuff like this, so that I can solve this kind of challenges myself?

**Comment by GaryAustin1** [comment]
These are all just Postgres trigger functions so Postgres docs or searching the web should show ideas.

Basically you can create 3 triggers (insert, update, delete) and call 3 functions OR you can have a trigger on all 3 call a single function and test which operation called the function.

"Example 43.4. A PL/pgSQL Trigger Function for Auditing" here is a good example of a single function.
https://www.postgresql.org/docs/current/plpgsql-trigger.html

**Comment by masnwilliams** [comment]
Here is what we use, you can just paste this into the SQL Editor or add it as a migration:
```
-- inserts a row into public.user
create function public.handle_new_user() 
returns trigger 
language plpgsql 
security definer set search_path = public
as $$
begin
  insert into public.user(id, email)
  values (new.id, new.email);

  insert into public.user_social(id, "userId")
  values (gen_random_uuid(), new.id);
  
  return new;
end;
$$;

-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

I was in the same position a few months ago so I hope this helps!
